src/tests/tests/debug.js

```diff
@@ -0,0 +1,74 @@
+(function() {
+  let debug = require("debug"),
+    beforeGetBadgerStorageObject = window.badger.storage.getBadgerStorageObject,
+    beforeGetAllOriginsForTab = window.badger.getAllOriginsForTab,
+    beforeCookiesGetAll = chrome.cookies.getAll;
+
+  QUnit.module("Debug", {
+    afterEach: () => {
+      window.badger.storage.getBadgerStorageObject = beforeGetBadgerStorageObject;
+      window.badger.getAllOriginsForTab = beforeGetAllOriginsForTab;
+      chrome.cookies.getAll = beforeCookiesGetAll;
+    },
+  });
+
+  QUnit.test("Debug Site", (assert) => {
+    // test data
+    let tab = {url: 'https://testmybadger.com/', id: 14},
+      testOrigins = ['foo.bar.com', 'bar.com', 'fake.stuff.com', 'more.data.com', 'otherstuff.com'].sort(),
+      inActionMap = ['foo.bar.com', 'bar.com', 'fake.stuff.com', 'more.data.com', /* omit otherstuff.com for testing */].sort(),
+      inSnitchMap = ['bar.com', 'stuff.com', /* omit more.data.com & otherstuff.com for testing */].sort();
+
+    // mock
+    window.badger.getAllOriginsForTab = function() {
+      return testOrigins;
+    };
+    window.badger.storage.getBadgerStorageObject = function(name) {
+      return {
+        getItemClones: function() {
+          let out = {};
+          if (name == 'action_map') {
+            inActionMap.forEach(d => {
+              out[d] = {not: 'empty'};
+            });
+            return out;
+          } else if (name == 'snitch_map') {
+            inSnitchMap.forEach(d => {
+              out[d] = ['not empty'];
+            });
+            return out;
+          }
+        }
+      };
+    };
+
+    let result = debug.debugTab(tab);
+    assert.deepEqual(result.info.origins.sort(), testOrigins.sort());
```

Comment:
Could we use assertion message parameters to describe what is being tested for each assertion? Something like:
```javascript
assert.deepEqual(
  result.info.origins.sort(),
  testOrigins.sort(),
  "output origins should match expected origins"
);
```